sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -673,6 +673,16 @@ case class AlterTableDropPartition(
   override def children: Seq[LogicalPlan] = child :: Nil
 }
 
+/**
+ * The logical plan of the ALTER TABLE ... RENAME TO PARTITION command.
+ */
+case class AlterTableRenamePartition(
+    child: LogicalPlan,
+    from: TablePartitionSpec,
+    to: TablePartitionSpec) extends Command {
```

Comment:
Please, use `PartitionSpec` https://github.com/apache/spark/blob/b112e2bfa619d028004cbc7fb8ec1363689729a7/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/v2ResolutionPlans.scala#L74 as in https://github.com/apache/spark/blob/0c19497222c26818ecdde527601c12c757acb4ad/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala#L667

Reply:
Updated. (I was going to introduce when implementing v2 version, but I guess it doesn't hurt to have it now).